src/usecases/validateCreateCompany.js

```diff
@@ -0,0 +1,28 @@
+import ValidationError from "../errors/validationError.js";
+
+function validateCreateCompany(company) {
+  const cpfCnpjRegex =
+    /(^\d{3}\.\d{3}\.\d{3}\-\d{2}$)|(^\d{2}\.\d{3}\.\d{3}\/\d{4}\-\d{2}$)/;
+
+  const listErrrors = [];
+
+  for (const key in company) {
+    if (!company[key]) {
+      listErrrors.push(`O campo ${key} é obrigatório`);
+    }
+  }
+
+  if (company.document) {
+    if (!cpfCnpjRegex.test(company.document)) {
+      listErrrors.push("O documento deve ser um CPF ou CNPJ válido");
+    }
+  }
```

Comment:
Acredito que você consiga resumir esses dois if em um if com duas expressões, algo como:

```suggestion
  if (company.document && !cpfCnpjRegex.test(company.document)) {
      listErrrors.push("O documento deve ser um CPF ou CNPJ válido");
  }
```

Faria sentido "quebrar" em ifs aninhados se você fosse fazer mais algum tratamento ou validação caso o campo `document` exista, além de checkar se é um valor válido. Como você apenas valida se o campo é válido, não precisa aninhar os IFs. 😉

Reply:
✍️